pages/wheelPage.js

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from "react";
+import { useEffect } from "react";
+import { StyledImage } from "@/components/StyledImage";
+import styled from "styled-components";
+import { StyleBox } from "@/components/StyledBox";
+import { StyledLink } from "@/components/StyledLink";
+import Link from "next/link";
+import { activities } from "@/lib/users/activities/activities";
+
+const ImageContainer = styled.div`
+  position: relative;
+  height: 25rem;
+  margin: 0rem;
+  padding: 1rem;
+`;
+
+const PlayButton = styled.button`
+  display: block;
+  width: 80px;
+  height: 80px;
+  border: none;
+  background: lightcyan;
+  font-weight: bold;
+  font-size: large;
+  border-radius: 50%;
+  box-shadow: 0px 5px 20px rgba(0, 0, 0, 0.2);
+  cursor: pointer;
+  position: fixed;
+  bottom: calc(5%);
+  left: 50%;
+  transform: translateX(-50%);
+`;
+const BoxLink = styled(Link)`
+  text-decoration: none;
+  color: black;
+`;
+
+const FixedBackLink = styled(StyledLink)`
+  position: fixed;
+  bottom: 30px;
+  left: 30px;
+  box-shadow: 0px 5px 20px rgba(0, 0, 0, 0.2);
+`;
+
+function Wheel() {
+  const [selectedActivity, setSelectedActivity] = useState(null);
+  const [spin, setSpin] = useState(false);
+
+  useEffect(() => {
+    if (spin) {
+      const spinTimeout = setTimeout(() => {
+        setSelectedActivity(
+          activities[Math.floor(Math.random() * activities.length)]
+        );
+        setSpin(false);
+      }, 5000);
+
+      return () => clearTimeout(spinTimeout);
+    }
+  }, [spin]);
+
+  const handleSpinClick = () => {
+    setSelectedActivity(null);
+    setSpin(true);
+  };
+  const renderPrize = (activity, index) => {
+    const activityStyle = {
+      transform: `rotate(${(360 / activities.length) * index}deg)`,
+      backgroundColor: index % 2 === 0 ? "#C1E1C1" : "#87CEFA",
+    };
+
+    const spinWheel = () => {
+      const randomIndex = Math.floor(Math.random() * activities.length);
+      setSelectedActivity(activities[randomIndex]);
+    };
+  };
+  return (
+    <div>
+      <div>{activities.map((activity, id) => renderPrize(activity, id))}</div>
```

Comment:
I don't think this code is doing anything…

You are calling the `renderPrize` function in a `map` during render. The `renderPrize` function does not return anything. (And also does not do anything with the two `const` it defines.)

So if you have 4 `activities` you will end up with the following in render:

```jsx
<div>{[undefined, undefined, undefined, undefined]}</div>
```

Which does not seem like anything you'd want.

![ahhh](https://media.giphy.com/media/FcuiZUneg1YRAu1lH2/giphy.gif)

```suggestion
  return (
    <div>
```